### PR TITLE
refactor(EditInPlace): add typescript types

### PR DESCRIPTION
**Pull request by anamikaanu96**
Contributes to #4498 

converted packages/ibm-products/src/components/EditInPlace/EditInPlace.js to packages/ibm-products/src/components/EditInPlace/EditInPlace.tsx

#### What did you change? 
packages/ibm-products/src/components/EditInPlace/EditInPlace.tsx

### CODE REVIEW

File: packages/ibm-products/src/components/EditInPlace/EditInPlace.tsx [hunk 1]
```diff
@@ -27,7 +33,101 @@ const defaults = {
...
+   * Can be passed either as one of tooltip options or as an object specifying cancel, edit and save separately
+   */
+  tooltipAlignment?: AlignPropType | Shape;
+  /**
+   * current value of the input
+   */
+  value: string;
+}
+
+export let EditInPlace = forwardRef<any, EditInplaceProps>(
```
Comment (matthewgallo) [inline packages/ibm-products/src/components/EditInPlace/EditInPlace.tsx:130]: Can you replace `any` with `HTMLDivElement`?

File: packages/ibm-products/src/components/EditInPlace/EditInPlace.tsx [hunk 2]
```diff
@@ -27,7 +33,101 @@ const defaults = {
...
+  /**
+   * label for cancel button
+   */
+  cancelLabel: string;
+  /**
+   * By default the edit icon is shown on hover only.
+   */
+  editAlwaysVisible?: boolean;
+  /**
+   /**
```
Comment (matthewgallo) [inline packages/ibm-products/src/components/EditInPlace/EditInPlace.tsx:62]: You have an extra `/**` on this line and the indentation is tad bit off. 🤔 I would have expected the precommit to have caught the indentation but running `yarn format` should fix it.

Comment (anamikaanu96) [reply packages/ibm-products/src/components/EditInPlace/EditInPlace.tsx]: pre-commit hook is running for me but not sure why its not catching this.  Also I noticed that when I do yarn format, its not formatting the tsx file. So I am doing format document.

Resolved 👍

**Review by matthewgallo — APPROVED** [review]

File: packages/ibm-products/src/components/EditInPlace/EditInPlace.tsx (hunk 2 above)
Comment (matthewgallo) [reply packages/ibm-products/src/components/EditInPlace/EditInPlace.tsx]: Ahh good catch and I see @szinta opened a PR to fix this as well 🥳

**Pull request merged**